test_runner/regress/test_remote_storage.py

```diff
@@ -627,9 +627,9 @@ def test_empty_branch_remote_storage_upload(
     new_branch_name = "new_branch"
     new_branch_timeline_id = env.neon_cli.create_branch(new_branch_name, "main", env.initial_tenant)
 
+    # FIXME: unsure why this is done
     with env.endpoints.create_start(new_branch_name, tenant_id=env.initial_tenant) as endpoint:
         wait_for_last_flush_lsn(env, endpoint, env.initial_tenant, new_branch_timeline_id)
-    wait_upload_queue_empty(client, env.initial_tenant, new_branch_timeline_id)
```

Comment:
Endpoint is needed to get the lsn which we await for being uploaded. There will be an exception if we fail to upload it in time. 

Other way to do that is to ask pageserver api for last record lsn and await for it

Reply:
Ok and now that on this PR we upload on `create_timeline` it can be safely removed. Didn't think of the uploading part. Feel free to unresolve if I was mistaken.

Reply:
Unresolving just to be sure that I understood correctly

>  it can be safely removed

Do you mean removing waiting? Or the whole endpoint creation? So in case only waiting is removed then there will be an assertion that remote consistent lsn will be past lsn we got from the endpoint, right?

Reply:
Yes, I was going to remove the endpoint creation as I already removed it in the `test_empty_branch_remote_storage_upload_on_restart`. But need to think about this, I'll ping when ready.

Reply:
> So in case only waiting is removed then there will be an assertion that remote consistent lsn will be past lsn we got from the endpoint, right?

This test case is for detach + attach restoring the both timelines.

The changes ensure this because `create_timeline` now awaits for `index_part.json` to be uploaded in both cases, and this will actually wait for the one initdb layer to be uploaded as well (during bootstrapping of the parent timeline) because layer is created and flushed, schedule `index_part.json` upload.

I think we might risk losing the L0 being turned into an image layer, if that still happens, but that's all. I am not sure if the original creation of endpoint + wait for uploads would had ensured that _directly_

So I think these lines could we removed completly, without any issues. No issues locally.

Reply:
Attempted to fix this in [d8804b1](https://github.com/neondatabase/neon/pull/4204/commits/d8804b146525c91a89a4c716b35dbb6050900e05). Test failures are unrelated `real_s3`, so should be good.

Reply:
Assuming ok, resolving.